level1a/handlers/level1a.py

```diff
@@ -153,89 +167,67 @@ def interpolate(dataframe: DataFrame, target: DatetimeIndex) -> DataFrame:
-        "_{i}.parquet"
-    ])
-
-
 def lambda_handler(event: Event, context: Context):
-    global last_date
     output_bucket = get_or_raise("OUTPUT_BUCKET")
-    rac_bucket = get_or_raise("RAC_BUCKET")
     platform_bucket = get_or_raise("PLATFORM_BUCKET")
     region = os.environ.get('AWS_REGION', "eu-north-1")
     s3 = pa.fs.S3FileSystem(region=region)
 
-    target_dataset = ds.dataset(output_bucket, filesystem=s3)
-
-    last_date = (
-        last_date
-        or get_last_date(
-            target_dataset,
-            Timestamp.now(tz=timezone.utc) - Timedelta(days=LOOKBACK_DAYS),
-        )
-        or get_last_date(
-            target_dataset,
-            DEFAULT_START,
-        )
-        or DEFAULT_START
-    )
+    try:
+        bucket, object = parse_event_message(event)
+    except InvalidMessage:
+        return {
+            'statusCode': HTTPStatus.NO_CONTENT,
+            'headers': {'Content-Type': 'application/json'},
+            'body': json.dumps({
+                'message': 'Failed to parse event, nothing to do.'
+            })
+        }
+
+    if not object.endswith(".parquet"):
+        return {
+            'statusCode': HTTPStatus.NO_CONTENT,
+            'headers': {'Content-Type': 'application/json'},
+            'body': json.dumps({
+                'message': f'{object} is not a parquet file, nothing to do.'
+            })
+        }
 
     rac_df = get_ccd_records(
-        rac_bucket,
-        last_date,
+        f"{bucket}/{object}",
         filesystem=s3,
     )
 
     min_time, max_time = get_search_bounds(rac_df.index)
 
     attitude_df = get_attitude_records(
         platform_bucket,
-        min_time,
-        max_time,
+        min_time - get_offset(ATTITUDE_FREQUENCY),
+        max_time + get_offset(ATTITUDE_FREQUENCY),
         filesystem=s3,
     )
     orbit_df = get_orbit_records(
         platform_bucket,
-        min_time,
-        max_time,
+        min_time - get_offset(ORBIT_FREQUENCY),
+        max_time + get_offset(ORBIT_FREQUENCY),
         filesystem=s3,
     )
+
+    if not covers(attitude_df.index, min_time, max_time):
+        raise DoesNotCover("Attitude data is missing timestamps")
+    if not covers(orbit_df.index, min_time, max_time):
+        raise DoesNotCover("Orbit data is missing timestamps")
+
     attitude_subset = interpolate(attitude_df, rac_df.index)
     orbit_subset = interpolate(orbit_df, rac_df.index)
-    partitioned_dates = get_partitioned_dates(rac_df.index)
     out_table = pa.Table.from_pandas(concat(
-        [rac_df, attitude_subset, orbit_subset, partitioned_dates],
+        [rac_df, attitude_subset, orbit_subset],
         axis=1,
     ))
 
-    pq.write_to_dataset(
-        table=[out_table],
-        root_path=output_bucket,
-        basename_template=get_filename(rac_df.index),
-        existing_data_behavior="overwrite_or_ignore",
+    pq.write_table(
+        out_table,
+        f"{output_bucket}/{object}",
```

Comment:
Note that partitioning in the RAC input bucket is _not_ on `EXPDate` but the `TMHeaderTime`, so there may be a small mismatch here, going forward.

Reply:
This does not repartition, just creates a single file at the same path as was read from but in the `output_bucket`

Reply:
Yes, just wanted you to be aware that, in case we want to use partitioning to make reading more efficient when looking up `EXPDate`, then this partitioning may at times be inaccurate.